atomos.framework/src/main/java/org/atomos/framework/base/AtomosCommands.java

```diff
@@ -86,29 +92,57 @@ private String getState(Bundle b) {
 		}
 	}
 
-	public void install(String name, String loaderType, File moduleDir) throws BundleException {
+	@Descriptor("Install a new layer")
+	public void install(@Descriptor("Name of the layer") String name,
+			@Descriptor("LoaderType of the layer [OSGI, SINGLE, MANY]") String loaderType,
+			@Descriptor("Directory where the AtomosBundles are stored") File moduleDir)
+					throws BundleException {
+
 		if (!moduleDir.isDirectory()) {
 			System.out.println("The specified path is not a directory: " + moduleDir.getAbsolutePath());
+			return;
+		}
+
+		final Optional<LoaderType> oLoaderType = Stream.of(LoaderType.values())
```

Comment:
I'm curious on the introduction of final local variables throughout.  I imagine this must be a practice you use in other projects?

Reply:
That was changes with an cleanup by the Save-Action. Settings came from am other Project. 
They use it because of Compiler optimisation (not sure) and they wouldn't "accidentally" try to modify a value....

That is why I asked for formatter an Checkstyles.

Reply:
My understanding is that the compiler detects any effectively final variables and treats them as final for optimizations such as that.  I think any such attempts to use final for local variable optimizations would likely be fruitless.  I can understand if you are trying to protect against accidental programming errors at compile time, but for simple for-each loops I think that is overboard.